homeassistant/components/scrape/sensor.py

```diff
@@ -97,45 +98,59 @@ async def async_setup_platform(
         rest = create_rest_data_from_config(hass, resource_config)
 
         coordinator = ScrapeCoordinator(hass, rest, SCAN_INTERVAL)
-        await coordinator.async_refresh()
-        if coordinator.data is None:
-            raise PlatformNotReady
 
-        sensor_config = config
-        template_config = vol.Schema(
+        sensor_config: ConfigType = config
+        template_config: ConfigType = vol.Schema(
```

Comment:
Looks great.
One more suggestion: it feels like some duplicate code could be removed...
Could we maybe generate a `sensors_config: list[tuple[ConfigType, ConfigType]]` and then process it to create the entities after `PlatformNotReady` check?

```suggestion
        sensors_config: list[tuple[ConfigType, ConfigType]] = [(config, vol.Schema(
```